pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -884,6 +879,19 @@ func (c *Reconciler) createTaskRun(ctx context.Context, taskRunName string, para
 	return c.PipelineClientSet.TektonV1().TaskRuns(pr.Namespace).Create(ctx, tr, metav1.CreateOptions{})
 }
 
+// handleRunCreationError marks the PipelineRun as failed and returns a permanent error if the run creation error is not retryable
+func (c *Reconciler) handleRunCreationError(ctx context.Context, pr *v1.PipelineRun, err error) error {
+	if controller.IsPermanentError(err) {
+		pr.Status.MarkFailed(ReasonCreateRunFailed, err.Error())
+		return err
+	}
+	if apierrors.IsInvalid(err) || apierrors.IsBadRequest(err) {
```

Comment:
If this is an encountered but not complete set of errors that should not be retried, should we add a comment here clarifying it?

Reply:
done